docs/api-reference/supervisor-api.md

```diff
@@ -23,89 +23,3075 @@ sidebar_label: Supervisors
+    },
+    "context": null,
+    "suspended": false
+}
+  ```
+</details>
 
-`POST /druid/indexer/v1/supervisor/suspendAll`
+### Resume all Supervisors
 
-Suspend all supervisors at once.
+#### URL
+<code className="postAPI">POST</code> `/druid/indexer/v1/supervisor/resumeAll`
 
-`POST /druid/indexer/v1/supervisor/<supervisorId>/resume`
+Resumes all Supervisors. 
 
-Resume indexing tasks for a supervisor. Responds with updated SupervisorSpec.
+Note that this endpoint will return a "success" message even if there are no Supervisors or suspended Supervisors.
 
-`POST /druid/indexer/v1/supervisor/resumeAll`
+#### Responses
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--200 SUCCESS-->
+<br/>
+*Successfully resumed all Supervisors* 
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+---
+
+#### Sample request
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--cURL-->
+```shell
+curl --request POST "http://<ROUTER_IP>:<ROUTER_PORT>/druid/indexer/v1/supervisor/resumeAll"
+```
+<!--HTTP-->
+```HTTP
+POST /druid/indexer/v1/supervisor/resumeAll HTTP/1.1
+Host: http://<ROUTER_IP>:<ROUTER_PORT>
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample response
+
+<details>
+  <summary>Click to show sample response</summary>
+
+  ```json
+{
+    "status": "success"
+}
+  ```
+</details>
+
+### Reset a Supervisor
+
+#### URL
+<code className="postAPI">POST</code> `/druid/indexer/v1/supervisor/{supervisorId}/reset`
+
+Resets the specified supervisor. This endpoint clears stored offsets, causing the Supervisor to start reading offsets from either the earliest or the latest offsets in Kafka. It kills and recreates active tasks to read from valid offsets. 
+
+Use this endpoint for recovering when the Supervisor stops due to missing offsets. Use this endpoint with caution as it may result in skipped, resulting in data loss, or duplicate data. 
+
+#### Responses
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--200 SUCCESS-->
+<br/>
+*Successfully reset Supervisor* 
+<!--404 NOT FOUND-->
+<br/>
+*Invalid Supervisor ID* 
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+---
+
+#### Sample request
+
+The following example resets the `social_media` Supervisor. 
+
+<!--DOCUSAURUS_CODE_TABS-->
+
+<!--cURL-->
+```shell
+curl --request POST "http://<ROUTER_IP>:<ROUTER_PORT>/druid/indexer/v1/supervisor/social_media/reset"
+```
+<!--HTTP-->
+```HTTP
+POST /druid/indexer/v1/supervisor/social_media/reset HTTP/1.1
+Host: http://<ROUTER_IP>:<ROUTER_PORT>
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+#### Sample response
+
+<details>
+  <summary>Click to show sample response</summary>
+
+  ```json
+{
+    "id": "social_media"
+}
+  ```
+</details>
+
+### Terminate a Supervisor
+
+#### URL 
+<code className="postAPI">POST</code> `/druid/indexer/v1/supervisor/{supervisorId}/terminate`
+
+Terminate a Supervisor of the provided ID and its associated indexing tasks, triggering the publishing of their segments. When terminated, a tombstone marker will be placed in the database to prevent reloading on restart. 
```

Comment:
```suggestion
Terminates a supervisor and its associated indexing tasks, triggering the publishing of their segments. When terminated, a tombstone marker is placed in the database to prevent reloading on restart. 
```